src/App.jsx

```diff
@@ -14,45 +18,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const goodsWithProperty = goodsFromServer.map((good, i) => ({
+  id: i + 1,
+  name: good,
+  length: good.length,
+}));
+
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+
+function prepeareGoods(goods, sortName, reverse) {
+  let prepearedGoods = [...goods];
+
+  if (sortName) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortName) {
+        case SORT_BY_NAME:
+          return good1.name.localeCompare(good2.name);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState('');
```

Comment:
```suggestion
  const [isReversed, setIsReverse]d = useState(false);
```
it is can be boolean state 
and change prevState after onClick on reverse button 
```
onClick=(() => setIsReversed(prevIsReversed => !prevIsReversed))
```

Reply:
It's interesting idea. Done!